python/ray/tune/trial.py

```diff
@@ -443,6 +445,22 @@ def last_result(self) -> dict:
     def last_result(self, val: dict):
         self._last_result = val
 
+    @property
+    def logdir(self):
+        if not self.relative_logdir:
+            return None
+        return str(Path(self.local_dir).joinpath(self.relative_logdir))
+
+    @logdir.setter
+    def logdir(self, logdir):
+        relative_logdir = Path(logdir).relative_to(self.local_dir)
+        assert str(relative_logdir).find("..") == -1
```

Comment:
Because this can potentially be set by a user/developer, let's not assert here but raise a ValueError if this is not the case

Reply:
Done

Reply:
Also changed the `test_trial_relative_logdir:testDotsInLogdir` accordingly